### ISSUE TITLE
[MJAVADOC-338] When includeDependencySources=true the javadoc plugin should not download the internet every time it is run

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#638]
**[Geoffrey De Smet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ge0ffrey.desmet)** opened **[MJAVADOC-338](https://issues.apache.org/jira/browse/MJAVADOC-338?redirect=false)** and commented

This is my configuration:

```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-javadoc-plugin</artifactId>
  <executions>
    <execution>
      <phase>package</phase>
      <goals>
        <goal>javadoc</goal>
      </goals>
    </execution>
  </executions>
  <configuration>
    <includeDependencySources>true</includeDependencySources>
    <includeTransitiveDependencySources>true</includeTransitiveDependencySources>
    <dependencySourceIncludes>
      <dependencySourceInclude>org.drools:*</dependencySourceInclude>
    </dependencySourceIncludes>
  </configuration>
</plugin>
```

the javadoc plugins downloads a very long list of dependencies every time it is run:

```
gdesmet@gdesmetRedHat2010 droolsjbpm-uber-distribution [master *] $ mvn javadoc:javadoc
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Drools and jBPM uber distribution 5.4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] >>> maven-javadoc-plugin:2.8:javadoc (default-cli) @ droolsjbpm-uber-distribution >>>
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0:enforce (default) @ droolsjbpm-uber-distribution ---
[INFO] 
[INFO] <<< maven-javadoc-plugin:2.8:javadoc (default-cli) @ droolsjbpm-uber-distribution <<<
[INFO] 
[INFO] --- maven-javadoc-plugin:2.8:javadoc (default-cli) @ droolsjbpm-uber-distribution ---
[WARNING] Missing POM for stax:stax-ri:jar:1.0
[WARNING] Missing POM for javax.ejb:ejb:jar:3.0
[WARNING] Missing POM for com.sun:tools:jar:javadoc-resources:1.5.0
[WARNING] Missing POM for woodstox:wstx-asl:jar:3.2.2
[WARNING] Missing POM for org.springframework.osgi:log4j.osgi:jar:1.2.15-SNAPSHOT
[WARNING] Missing POM for javax.servlet.jsp.jstl:jstl:jar:1.2
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.beans/3.0.0.RELEASE/org.springframework.beans-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.beans/3.0.0.RELEASE/org.springframework.beans-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.beans/3.0.0.RELEASE/org.springframework.beans-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.beans:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.asm/3.0.0.RELEASE/org.springframework.asm-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.asm/3.0.0.RELEASE/org.springframework.asm-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.asm/3.0.0.RELEASE/org.springframework.asm-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.asm:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.core/3.0.0.RELEASE/org.springframework.core-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.core/3.0.0.RELEASE/org.springframework.core-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.core/3.0.0.RELEASE/org.springframework.core-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.core:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.aop/3.0.0.RELEASE/org.springframework.aop-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.aop/3.0.0.RELEASE/org.springframework.aop-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.aop/3.0.0.RELEASE/org.springframework.aop-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.aop:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.aspects/3.0.0.RELEASE/org.springframework.aspects-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.aspects/3.0.0.RELEASE/org.springframework.aspects-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.aspects/3.0.0.RELEASE/org.springframework.aspects-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.aspects:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.context/3.0.0.RELEASE/org.springframework.context-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.context/3.0.0.RELEASE/org.springframework.context-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.context/3.0.0.RELEASE/org.springframework.context-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.context:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.context.support/3.0.0.RELEASE/org.springframework.context.support-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.context.support/3.0.0.RELEASE/org.springframework.context.support-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.context.support/3.0.0.RELEASE/org.springframework.context.support-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.context.support:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/springframework/org.springframework.expression/3.0.0.RELEASE/org.springframework.expression-3.0.0.RELEASE.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/springframework/org.springframework.expression/3.0.0.RELEASE/org.springframework.expression-3.0.0.RELEASE.pom
Downloading: http://repo1.maven.org/maven2/org/springframework/org.springframework.expression/3.0.0.RELEASE/org.springframework.expression-3.0.0.RELEASE.pom
[WARNING] Missing POM for org.springframework:org.springframework.expression:jar:javadoc-resources:3.0.0.RELEASE
Downloading: http://repository.jboss.org/nexus/content/groups/public/net/sourceforge/jexcelapi/com.springsource.jxl/2.6.6/com.springsource.jxl-2.6.6.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/net/sourceforge/jexcelapi/com.springsource.jxl/2.6.6/com.springsource.jxl-2.6.6.pom
Downloading: http://repo1.maven.org/maven2/net/sourceforge/jexcelapi/com.springsource.jxl/2.6.6/com.springsource.jxl-2.6.6.pom
[WARNING] Missing POM for net.sourceforge.jexcelapi:com.springsource.jxl:jar:javadoc-resources:2.6.6
Downloading: http://repository.jboss.org/nexus/content/groups/public/com/thoughtworks/xstream/com.springsource.com.thoughtworks.xstream/1.3.1/com.springsource.com.thoughtworks.xstream-1.3.1.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/com/thoughtworks/xstream/com.springsource.com.thoughtworks.xstream/1.3.1/com.springsource.com.thoughtworks.xstream-1.3.1.pom
Downloading: http://repo1.maven.org/maven2/com/thoughtworks/xstream/com.springsource.com.thoughtworks.xstream/1.3.1/com.springsource.com.thoughtworks.xstream-1.3.1.pom
[WARNING] Missing POM for com.thoughtworks.xstream:com.springsource.com.thoughtworks.xstream:jar:javadoc-resources:1.3.1
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/joda/com.springsource.org.joda.time/1.6.0/com.springsource.org.joda.time-1.6.0.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/joda/com.springsource.org.joda.time/1.6.0/com.springsource.org.joda.time-1.6.0.pom
Downloading: http://repo1.maven.org/maven2/org/joda/com.springsource.org.joda.time/1.6.0/com.springsource.org.joda.time-1.6.0.pom
[WARNING] Missing POM for org.joda:com.springsource.org.joda.time:jar:javadoc-resources:1.6.0
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/antlr/com.springsource.org.antlr.runtime/3.1.3/com.springsource.org.antlr.runtime-3.1.3.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/antlr/com.springsource.org.antlr.runtime/3.1.3/com.springsource.org.antlr.runtime-3.1.3.pom
Downloading: http://repo1.maven.org/maven2/org/antlr/com.springsource.org.antlr.runtime/3.1.3/com.springsource.org.antlr.runtime-3.1.3.pom
[WARNING] Missing POM for org.antlr:com.springsource.org.antlr.runtime:jar:javadoc-resources:3.1.3
Downloading: http://repository.jboss.org/nexus/content/groups/public/org/antlr/com.springsource.org.antlr/3.1.3/com.springsource.org.antlr-3.1.3.pom
Downloading: https://repository.jboss.org/nexus/content/repositories/deprecated/org/antlr/com.springsource.org.antlr/3.1.3/com.springsource.org.antlr-3.1.3.pom
Downloading: http://repo1.maven.org/maven2/org/antlr/com.springsource.org.antlr/3.1.3/com.springsource.org.antlr-3.1.3.pom
[WARNING] Missing POM for org.antlr:com.springsource.org.antlr:jar:javadoc-resources:3.1.3
```

---

**Affects:** 2.8

**Issue Links:**
- [MJAVADOC-525](https://issues.apache.org/jira/browse/MJAVADOC-525) When includeDependencySources=true the javadoc plugin should not download the internet every time it is run
 (_**"is cloned by"**_)
- [MJAVADOC-525](https://issues.apache.org/jira/browse/MJAVADOC-525) When includeDependencySources=true the javadoc plugin should not download the internet every time it is run
 (_**"is duplicated by"**_)

2 votes, 7 watchers

**Comment by jira-importer** [comment]
**[Geoffrey De Smet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ge0ffrey.desmet)** commented

Notice how it always looks for "javadoc-resources" artifacts. But since I am not use those, I don't see the point.

**Comment by jira-importer** [comment]
**[Peter Hayes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hayesp25)** commented

I have the same comment as well.  I'd like to use this feature but the build performance cost is high.  In my case, I'm only using it to include 2 of my additional libraries but it still tries to access the javadoc-resources of every dependency of my project.  At a minimum it should only look for those javadoc-resources that will be included.

Here's a portion of my build output for reference:

```
[INFO] --- maven-javadoc-plugin:2.9:javadoc (published-api) @ uma-imdesktop ---
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/net/sf/ehcache/sizeof-agent/1.0.1/sizeof-agent-1.0.1.pom
[WARNING] Missing POM for net.sf.ehcache:sizeof-agent:jar:1.0.1
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/ibm/websphere/uow/6.0.2.17/uow-6.0.2.17.pom
[WARNING] Missing POM for com.ibm.websphere:uow:jar:6.0.2.17
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/stax/stax-ri/1.0/stax-ri-1.0.pom
[WARNING] Missing POM for stax:stax-ri:jar:1.0
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/slf4j/slf4j-api/1.6.1/slf4j-api-1.6.1-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/log4j/log4j/1.2.15/log4j-1.2.15-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/slf4j/slf4j-log4j12/1.6.1/slf4j-log4j12-1.6.1-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/fimt/asts/ws/frf-handlers/3.0.0/frf-handlers-3.0.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/fimt/asts/ws/frf-identity/3.0.0/frf-identity-3.0.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/fimt/asts/ws/frf-utils/3.0.0/frf-utils-3.0.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/fimt/asts/ws/config-ws-stubs/1.0.0/config-ws-stubs-1.0.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/com/fimt/asts/frf/maven-archetype-frf-contract-first/3.0.0/maven-archetype-frf-contract-first-3.0.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/springframework/spring-aop/3.2.1.RELEASE/spring-aop-3.2.1.RELEASE-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/aopalliance/aopalliance/1.0/aopalliance-1.0-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/springframework/spring-beans/3.2.1.RELEASE/spring-beans-3.2.1.RELEASE-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/springframework/spring-core/3.2.1.RELEASE/spring-core-3.2.1.RELEASE-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/codehaus/jackson/jackson-core-asl/1.9.4/jackson-core-asl-1.9.4-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/codehaus/jackson/jackson-mapper-asl/1.9.4/jackson-mapper-asl-1.9.4-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/springframework/spring-context/3.2.1.RELEASE/spring-context-3.2.1.RELEASE-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/org/springframework/spring-expression/3.2.1.RELEASE/spring-expression-3.2.1.RELEASE-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/commons-lang/commons-lang/2.4/commons-lang-2.4-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/commons-httpclient/commons-httpclient/3.1/commons-httpclient-3.1-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/commons-codec/commons-codec/1.4/commons-codec-1.4-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/javax/transaction/jta/1.1/jta-1.1-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/javax/xml/bind/jaxb-api/2.2/jaxb-api-2.2-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2-javadoc-resources.jar
Downloading: http://fimt-nexus-repository.fmrco.com/nexus/content/groups/public/javax/activation/activation/1.1/activation-1.1-javadoc-resources.jar
```

**Comment by jira-importer** [comment]
**[Sander Verhagen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sverhagen)** commented

I would say it has nothing to do with javadoc-resources.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

It looks like it's doing what you asked it to do: trying to resolve javadoc-resources bundles for all transitive drools dependencies. I suppose you might be able to refine the configuration better if you had a set of includes/excludes configurations specific to javadoc-resources, particularly since these aren't all that common. But the log you posted is just a long list of 404's, right? So, they're probably coming back fairly quickly, and IIRC the 404 result should be stored in your local repository (unless you're using -U to bypass that result caching).

**Comment by jira-importer** [comment]
**[Sander Verhagen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sverhagen)** commented

The specification of how `dependencySourceInclude` may then leave room for different expectations. My expectation was that the included sources would apply transitively. So who configures `<dependencySourceInclude>org.drools:*</dependencySourceInclude>` will only get `org.drools` dependencies, and not transitive dependencies, unless they're also in `org.drools`.

The use case I'm pursuing is the following: `my.company:api-1` depends on `my.company:common-api`. I want to generate aggregated Javadocs that include `api-1` and `common-api`, starting my build from `api-1`. Since `common-api` depends on a bunch of other dependencies, these are downloaded. Funny thing is, the Javadocs only include `api-1` and `common-api` as per my `dependencySourceInclude`. But then why does it need to download so many dependencies?

Hope this explanation is helpful.

**Comment by jira-importer** [comment]
**[Geoffrey De Smet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ge0ffrey.desmet)** commented

`@JohnCasey` Although I only want to include the drools dependency's javadoc-resources, it's also downloading those of slf4j, springframework, etc. Any way to stop that?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This issue has been auto closed because it has been inactive for a long period of time. If you think this issue still applies, retest your problem with the most recent version of Maven and the affected component, reopen and post your results.

**Comment by jira-importer** [comment]
**[Donatello](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=donatello)** commented

The issue is still there (using maven-javadoc-plugin:3.0.1) with _includeDependencySources = true_ and **includeTransitiveDependencySources = false** :

```xml
<build>
    <plugins>
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-javadoc-plugin</artifactId>
            <executions>
                <execution>
                    <id>aggregate-javadoc</id>
                    <goals>
                        <goal>jar</goal>
                    </goals>
                    <configuration>
                        <outputDirectory>target/apidocs</outputDirectory>
                        <attach>true</attach>
                        <debug>true</debug>
                        <verbose>true</verbose>
                        <includeDependencySources>true</includeDependencySources>
                        <includeTransitiveDependencySources>false</includeTransitiveDependencySources>
                        <stylesheetfile>${basedir}/src/main/javadoc/javadoc_stylesheet.css</stylesheetfile>
                        <detectOfflineLinks>false</detectOfflineLinks>
                        <links>
                            <link>http://docs.oracle.com/javase/8/docs/api/</link>
                        </links>
                        <excludePackageNames>.internal:.config</excludePackageNames>
                        <sourceFileExcludes>
                            <exclude>com/company/core/Check*.java</exclude>
                        </sourceFileExcludes>
                    </configuration>
                </execution>
            </executions>
        </plugin>
    </plugins>
</build>
<dependencies>
    <!-- framework -->
    <dependency>
        <groupId>com.company</groupId>
        <artifactId>fwk-library-1</artifactId>
    </dependency>
    <dependency>
        <groupId>com.company</groupId>
        <artifactId>fwk-library-2</artifactId>
    </dependency>
    <!-- test materials -->
    <dependency>
        <groupId>com.company</groupId>
        <artifactId>fwk-commons</artifactId>
        <type>test-jar</type>
    </dependency>
</dependencies>
```

 

Which results in a veeery long list of `javadoc-resources.jar` download attempts:

```none
[INFO] --- maven-javadoc-plugin:3.0.1:jar (aggregate-javadoc) @ im-javadoc ---                                                                                                                                 
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/apache/activemq/activemq-openwire-legacy/5.15.7/activemq-openwire-legacy-5.15.7-javadoc-resources.jar                          
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/neo4j/driver/neo4j-java-driver/1.7.2/neo4j-java-driver-1.7.2-javadoc-resources.jar                                             
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/aspectj/aspectjrt/1.8.9/aspectjrt-1.8.9-javadoc-resources.jar                                                                  
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-server/9.4.11.v20180605/jetty-server-9.4.11.v20180605-javadoc-resources.jar                                
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0-javadoc-resources.jar                                                
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-http/9.4.11.v20180605/jetty-http-9.4.11.v20180605-javadoc-resources.jar                                    
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-io/9.4.11.v20180605/jetty-io-9.4.11.v20180605-javadoc-resources.jar                                        
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-webapp/9.4.11.v20180605/jetty-webapp-9.4.11.v20180605-javadoc-resources.jar                                
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-xml/9.4.11.v20180605/jetty-xml-9.4.11.v20180605-javadoc-resources.jar                                      
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-util/9.4.11.v20180605/jetty-util-9.4.11.v20180605-javadoc-resources.jar                                    
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-servlet/9.4.11.v20180605/jetty-servlet-9.4.11.v20180605-javadoc-resources.jar                              
Downloading: http://vbet0008g.gre.hpecorp.net:8080/nexus/content/groups/public/org/eclipse/jetty/jetty-security/9.4.11.v20180605/jetty-security-9.4.11.v20180605-javadoc-resources.jar
[...]
```

 

This process can take almost one hour ....

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [fd0c42a5426e2c6a97fd866548fbaba39106a13f](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=fd0c42a5426e2c6a97fd866548fbaba39106a13f)
